app/Request.php

```diff
@@ -151,12 +152,13 @@ public function get($key, $value = '')
 	 */
 	public function getByType($key, $type = 'Standard')
 	{
-		if (isset($this->purifiedValuesByType[$key])) {
-			return $this->purifiedValuesByType[$key];
+		if (isset($this->purifiedValuesByType[$key][$type])) {
+			return $this->purifiedValuesByType[$key][$type];
 		}
 		if (isset($this->rawValues[$key])) {
-			return $this->purifiedValuesByType[$key] = Purifier::purifyByType($this->rawValues[$key], $type);
+			return $this->purifiedValuesByType[$key][$type] = Purifier::purifyByType($this->rawValues[$key], $type);
```

Comment:
Clean purifiedValuesByType when set value

Reply:
clean $this->purifiedValuesByType[$key] in set() method